uber/model_checks.py

```diff
@@ -190,6 +190,27 @@ def printed_badge_deadline(attendee):
 
 
 @validation.Attendee
+def printed_badge_change(attendee):
+    badge_name_changes_allowed = True
+
+    # this is getting kinda messy and we probably need to rework the entire concept of "printed badge deadline".
+    # right now we want to:
+    # 1) allow supporters to change their badge names until c.SUPPORTER_DEADLINE
+    # 2) allow staff to change their badge names until c.PRINTED_BADGE_DEADLINE
+    #
+    # this implies that we actually have two different printed badge deadlines: 1 for staff, 1 for supporters.
+    # we might just want to make that explicit.
+    if attendee.badge_type == c.STAFF_BADGE and c.AFTER_PRINTED_BADGE_DEADLINE:
```

Comment:
Let's check for `in c.PREASSIGNED_ BADGE_ TYPES` instead of just looking for staff badges.